.github/workflows/update-docs.yml

```diff
@@ -0,0 +1,35 @@
+name: "Push New Tagged Version Docs Build" # Build on a new version tag
+on:
+  - push:
+    branches:
+      - "master"
+    tags:
+      - "[0-9]+.[0-9]+.[0-9]+"
```

Comment:
Were you able to test run this action? Perhaps w/ a dummy `gh-pages-test` branch?

Reply:
We could maybe build in consistent "staging" / test support by making the branch to which the documentation is pushed [conditional on the tag format](https://stackoverflow.com/a/58178121) -- i.e., we could just have a persistent `gh-pages-staging` or `gh-pages-test` branch to which pre-release tag doc builds are pushed (i.e., if the tag is an alpha or beta tag)

Reply:
I have not tested this. I'll try with a staging branch.

Reply:
Tested on my fork and confirmed to work with comments in the workflow to describe what GH access TOKEN permissions are required